### PR TITLE
feat(lib): qri.get_dataset function for updates

### PR DESCRIPTION
**Pull request by dustmop**
When a user calls ```qri update ....``` with a transformation, they can use qri.get_dataset() in skylark to get the body of the previous dataset.

### CODE REVIEW

File: lib/qri/qri.go [hunk 1]
```diff
@@ -71,6 +76,24 @@ func (m *Module) GetConfig(thread *skylark.Thread, _ *skylark.Builtin, args skyl
...
+func (m *Module) GetDataset(thread *skylark.Thread, _ *skylark.Builtin, args skylark.Tuple, kwargs []skylark.Tuple) (skylark.Value, error) {
+	if m.infile == nil {
+		return skylark.None, fmt.Errorf("qri.get_dataset failed: no DataFile")
+	}
+	bytes, err := ioutil.ReadAll(m.infile)
+	if err != nil {
+		return skylark.None, err
+	}
+	var object interface{}
+	err = json.Unmarshal(bytes, &object)
```
Comment (b5) [inline lib/qri/qri.go]: Can't assume provided data is JSON. Instead try:
```golang
var (
  bytes []byte
  err error
  object interface{}
)

if ds.Structure.Format == dataset.JSONDataFormat {
  bytes, err = ioutil.ReadAll(m.infile)
  if err != nil {
    return skylark.None, err
  }
} else {
  r, err := dsio.NewReader(m.ds.Structure, m.infile)
  // if err != nil ...
  buf, err := dsio.NewEntryBuffer(&dataset.Structure{Format : dataset.JSONDataFormat, Schema: m.ds.Structure.Schema})
  
  err := dsio.EachEntry(r, func(_ int, ent Entry, err error) error {
    if err != nil {
      return err
    }
    return buf.WriteEntry(ent)
  })
  // if err != nil ...
  bytes = buf.Bytes()
}

err = json.Unmarshal(bytes, &object)
// ...
```
The above is super verbose. I'll open an issue...

Comment (b5) [reply lib/qri/qri.go]: qri-io/dataset#127

**Comment by dustmop** [comment]
Updated to be get_body instead of get_dataset, and use the new Copy convenience function.

File: lib/lib.go [hunk 2]
```diff
@@ -122,6 +122,8 @@ func Marshal(data interface{}) (v skylark.Value, err error) {
     v = skylark.Bool(x)
   case string:
     v = skylark.String(x)
+  case int:
+    v = skylark.MakeInt(x)
   case float64:
     // TODO - ints?
```
Comment (b5) [inline lib/lib.go]: Thanks to your addition, we can nix this TODO in a future PR

File: lib/qri/entry_writer.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+import (
+	"github.com/google/skylark"
+	"github.com/qri-io/skytf/lib"
+	"github.com/qri-io/dataset"
+	"github.com/qri-io/dataset/dsio"
+)
+
+// SkylarkEntryWriter creates a skylark.Value as an EntryWriter
+type SkylarkEntryWriter struct {
```
Comment (b5) [inline lib/qri/entry_writer.go]: phenom. We should get tests up for this guy soon, but this implementation looks great.

File: lib/qri/qri.go [hunk 4]
```diff
@@ -14,24 +16,26 @@ import (
 const ModuleName = "qri.sky"
 
 // NewModule creates a new qri module instance
-func NewModule(ds *dataset.Dataset, secrets map[string]interface{}) *Module {
-	return &Module{ds: ds, secrets: secrets}
+func NewModule(ds *dataset.Dataset, secrets map[string]interface{}, infile cafs.File) *Module {
```
Comment (b5) [inline lib/qri/qri.go]: I'll have to clean this up in relation to #2, but the addition of `infile` is obviously needed

File: lib/qri/entry_writer.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+// Value returns the underlying skylark.Value
+func (w *SkylarkEntryWriter) Value() skylark.Value {
+	return w.Object
+}
+
+// NewSkylarkEntryWriter returns a new SkylarkEntryWriter
+func NewSkylarkEntryWriter(st *dataset.Structure) (*SkylarkEntryWriter, error) {
+	// TODO: Better method of detecting array vs object.
+	if st.Schema == dataset.BaseSchemaObject {
```
Comment (b5) [inline lib/qri/entry_writer.go]: lol I should have caught this

**Pull request merged**